tests/poll.rs

```diff
@@ -201,10 +208,112 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+// Testing on Windows only where this must fail by design, on other platforms
+// behavior is undefined.
+#[test]
+#[cfg(target_os = "Windows")]
+fn register_deregister_multiple_poll_instances() {
+    /*
+    This test checks the following register/deregister contrain:
+    A handle can be passed back to `register` after it has been
```

Comment:
Is this true for `reregister` as well? If so, should that be test? If not, could there be a comment explaining why?

Reply:
I've added a test for the following sequence `poll1.register(listener); poll2.reregister(listener);`. Is this what you suggested? Thanks